test/parallel/test-http-expect-handling.js

```diff
@@ -0,0 +1,35 @@
+// Spec documentation http://httpwg.github.io/specs/rfc7231.html#header.expect
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const http = require('http');
+
+const server = http.createServer((req, res) => {
+  throw new Error('This shouldn\'t have been called');
+}).listen(common.PORT);
+
+server.on('listening', () => {
+  const options = {
+    port: common.PORT,
+    headers: { 'Expect': 'meoww' }
+  };
+
+  const optionsGood = {
+    port: common.PORT,
+    headers: { 'Expect': '100-continue' }
+  };
+
+  http.request(options).on('response', (res) => {
+    const statusErrorMsg = `Final status code was ${res.statusCode}, not 417.`;
+
+    assert.equal(417, res.statusCode, statusErrorMsg);
+
+    server.on('checkExpectation', common.mustCall((req, res) => {
+      res.end();
+    }));
+
+    http.request(options).on('response', (res) => {
+      process.exit();
```

Comment:
we generally try to avoid explicitly calling `process.exit()` in tests. Could you instead just close the server (which should make the process exit on it's own)?